### ISSUE TITLE
TXT as expression replacement in INTERFACEs

### ISSUE THREAD

**Issue by RieksJ** [AmpersandTarski/Ampersand#792]
Currently, the syntax of a 'line' in an INTERFACE/BOX is roughly as follows:
~~~
   <label><colon><expression><crud><optionalviewstuff>
~~~
There are many cases where we want `<expression>` to be a text. In VIOLATIONs and VIEWs, we can already use the `TXT` keyword to specify such texts. We (@Michiel-s and I) really want to have that in INTERFACE/BOXes, too. This would allow a BOX specification such as below to be acceptable:
~~~
    BOX <DIV> [  "fixed text": TXT "Please be warned: ",  "warningtext": warningText cRud ] 
~~~
Code to be generated should be similar to what is generated for VIOLATIONs and VIEWs.

**Comment by hanjoosten** [comment]
I like the idea. However, I think a slightly different approach could be more practical AND easier to implement. In these cases, I see no need for the `<label><colon>` part. 

So a BOX specification like this would be acceptable:

~~~
    BOX <DIV> [  TXT "Please be warned: "
                        ,  "warningtext": warningText ] 
~~~

@Michiel-s , we need to discuss the underlying json.

**Comment by hanjoosten** [comment]
@Michiel-s , I have created a branch for this issue. The parsing and typechecking part is done. However, now I struggle to what to generate. What template should we use? How exactly would such a thing work? 

Currently, the branch compiles. There are a few modifications in the generated JSON:

- The list of objects in `ifcObjects` used to be of a single type. Now, this is no longer true. This is modelled by adding a `tag` field. The values the tag-field can have is either `"ObjExpression"` or `"ObjText"`. 
- For each object a view is generated from a template. Do we want this mechanism too for the TXT field?

At this moment, if you add a TXT field to an interface, you will see a fatal message. That message shows the place in the code where still work is to be done. Maybe time for some pair-programming?

**Comment by RieksJ** [comment]
Risking myself to be considered somewhat pushy, here's something I would have liked as well:
~~~
r :: A * B
VIEW MyViewOnB:  B  { <some complex view for concept B> }

VIEW MyViewOnA: A { <stuff>, "b", **r <MyViewOnB>**, <more stuff>
~~~
In a similar fashion, such constructs coult then appear in VIOLATION Texts, e.g.
~~~
VIOLATION (TXT "There is a violation on pair (", **SRC I[A] <MyViewOnA>**, TXT ", ", **TGT I[B] <MyViewOnB>**, TXT ")")
~~~
Is that simple enough to take along?

**Comment by hanjoosten** [comment]
An interface contains a Box, which contains a comma-separated list of box-items. Until now, a box-item could have a single form only:

- `<label>` `:` `<expression>` `<cruds>?`  `<viewTemplate>?` `<subInterface>?` 

With this enhancement implemented, a box-item can have one more form:

- `<label>` `:` `TXT` `<string>`

**Comment by hanjoosten** [comment]
@Michiel-s : I added the semicolon, as you requested. As far as I can tell, the following things remain to be done:

- [ ] Add an HTML template for these TXT lines.
- [ ] Modify GenFrontend.hs so it will use this HTML template
- [ ] Update the frontend, so that the .json is handled correctly

**Comment by hanjoosten** [comment]
@RieksJ : The VIEW statement has to be discussed with @stefjoosten and @sjcjoosten , for there are ideas to integrate VIEW and INTERFACE, VIOLATION

**Comment by Michiel-s** [comment]
Will be available after merge of #869
